frameworks/C++/ffead-cpp/te-benchmark-um/src/TeBkUm.cpp

```diff
@@ -0,0 +1,312 @@
+/*
+ * TeBkUmUm.cpp
+ *
+ *  Created on: 03-Feb-2020
+ *      Author: sumeetc
+ */
+#include "TeBkUm.h"
+
+int TeBkUmFortune::getId() const {
+	return id;
+}
+
+void TeBkUmFortune::setId(int id) {
+	this->id = id;
+}
+
+const std::string& TeBkUmFortune::getMessage() const {
+	return message;
+}
+
+void TeBkUmFortune::setMessage(const std::string& message) {
+	this->message = message;
+}
+
+TeBkUmFortune::TeBkUmFortune() {
+	id = 0;
+}
+
+TeBkUmFortune::~TeBkUmFortune() {
+}
+
+bool TeBkUmFortune::operator < (const TeBkUmFortune& other) const {
+	return message.compare(other.message)<0;
+}
+
+TeBkUmMessage::~TeBkUmMessage() {
+}
+
+const std::string& TeBkUmMessage::getMessage() const {
+	return message;
+}
+
+void TeBkUmMessage::setMessage(const std::string& message) {
+	this->message = message;
+}
+
+const std::string TeBkUmRouter::HELLO_WORLD = "Hello, World!";
+std::string TeBkUmRouter::WORLD = "world";
+
+void TeBkUmRouter::db(TeBkUmWorld& w) {
+	DataSourceInterface* sqli = DataSourceManager::getImpl();
+	int rid = rand() % 10000 + 1;
+	GenericObject id;
+	id << rid;
+	try {
+		w = sqli->get<TeBkUmWorld>(id);
+		delete sqli;
+	} catch(const std::exception& e) {
+		delete sqli;
+		throw e;
+	}
+}
+
+void TeBkUmRouter::queries(const char* q, int ql, std::vector<TeBkUmWorld>& wlst) {
+	int queryCount = 0;
+	strToNum(q, ql, queryCount);
+	if(queryCount<1)queryCount=1;
+	else if(queryCount>500)queryCount=500;
+
+	DataSourceInterface* sqli = DataSourceManager::getImpl();
+
+	try {
+		sqli->startSession(&WORLD);
+		for (int c = 0; c < queryCount; ++c) {
+			int rid = rand() % 10000 + 1;
+			GenericObject id;
+			id << rid;
+			TeBkUmWorld w = sqli->get<TeBkUmWorld>(id);
+			wlst.push_back(w);
+		}
+		sqli->endSession();
+		delete sqli;
+	} catch(const std::exception& e) {
+		delete sqli;
+		throw e;
+	}
+}
+
+void TeBkUmRouter::updates(const char* q, int ql, std::vector<TeBkUmWorld>& wlst) {
+	int queryCount = 0;
+	strToNum(q, ql, queryCount);
+	if(queryCount<1)queryCount=1;
+	else if(queryCount>500)queryCount=500;
+
+	DataSourceInterface* sqli = DataSourceManager::getImpl();
+
+	try {
+		sqli->startSession(&WORLD);
+		for (int c = 0; c < queryCount; ++c) {
+			int rid = rand() % 10000 + 1;
+			GenericObject id;
+			id << rid;
+			TeBkUmWorld w = sqli->get<TeBkUmWorld>(id);
+			int newRandomNumber = rand() % 10000 + 1;
+			if(w.getRandomNumber() == newRandomNumber) {
+				newRandomNumber -= 1;
```

Comment:
If the randomNumber is 1, this will set it to 0 and cause an error in the verification. This would be rare, but it could happen during an official run.